snuba/query/processors/logical/filter_in_select_optimizer.py

```diff
@@ -1,234 +1,145 @@
 import logging
+from copy import deepcopy
 
 from snuba import environment
 from snuba.query.composite import CompositeQuery
 from snuba.query.conditions import binary_condition
 from snuba.query.data_source.simple import Entity as QueryEntity
 from snuba.query.expressions import (
+    Argument,
     Column,
     CurriedFunctionCall,
-    Expression,
+    ExpressionVisitor,
     FunctionCall,
+    Lambda,
     Literal,
     SubscriptableReference,
 )
 from snuba.query.logical import Query as LogicalQuery
 from snuba.state import get_int_config
 from snuba.utils.metrics.wrapper import MetricsWrapper
 
-"""
-Domain maps from a property to the specific values that are being filtered for. Ex:
-    sumIf(value, metric_id in [1,2,3])
-        Domain = {metric_id: {1,2,3}}
-    sumIf(value, metric_id=5 and status=200) / sumIf(value, metric_id=5 and status=400)
-        Domain = {
-            metric_id: {5},
-            status: {200, 400}
-        }
-"""
-Domain = dict[Column | SubscriptableReference, set[Literal]]
-
 metrics = MetricsWrapper(environment.metrics, "api")
 
 logger = logging.getLogger(__name__)
 
 
+class FindConditionalAggregateFunctionsVisitor(ExpressionVisitor[None]):
+    """
+    Search an expression for all conditional aggregate functions.
+
+    Might be able to use matchers.Pattern to extend this idea for general search but I was too lazy.
+    """
+
+    def __init__(self) -> None:
+        self._matches: list[FunctionCall | CurriedFunctionCall] = []
+
+    def get_matches(self) -> list[FunctionCall | CurriedFunctionCall]:
+        return deepcopy(self._matches)
```

Comment:
My copying was done so that when I put it in WHERE, changing SELECT wouldnt modify the one in WHERE.

Volo helped me see that the copying shouldnt be done in the visitor, but later on when i copy the one from SELECT into WHERE.